tests/tests.rs

```diff
@@ -0,0 +1,200 @@
+        Decode!(
+            &assert_reply(
+                self.env
+                    .execute_ingress_as(self.caller, self.evm_rpc_id, method, input,)
+                    .unwrap_or_else(|err| panic!(
+                        "error during update call to `{}()`: {}",
+                        method, err
+                    ))
+            ),
+            R
+        )
+        .unwrap()
+    }
+
+    fn call_query<R: CandidType + DeserializeOwned>(&self, method: &str, input: Vec<u8>) -> R {
+        Decode!(
+            &assert_reply(
+                self.env
+                    .query_as(self.caller, self.evm_rpc_id, method, input,)
+                    .unwrap_or_else(|err| panic!(
+                        "error during query call to `{}()`: {}",
+                        method, err
+                    ))
+            ),
+            R
+        )
+        .unwrap()
+    }
+
+    pub fn authorize(&self, principal: &PrincipalId, auth: Auth) {
+        self.call_update("authorize", Encode!(&principal.0, &auth).unwrap())
+    }
+
+    pub fn deauthorize(&self, principal: &PrincipalId, auth: Auth) {
+        self.call_update("deauthorize", Encode!(&principal.0, &auth).unwrap())
+    }
+
+    pub fn get_providers(&self) -> Vec<ProviderView> {
+        self.call_update("get_providers", Encode!().unwrap())
+    }
+
+    pub fn register_provider(&self, args: RegisterProviderArgs) -> u64 {
+        self.call_update("register_provider", Encode!(&args).unwrap())
+    }
+
+    pub fn authorize_caller(self, auth: Auth) -> Self {
+        self.as_controller().authorize(&self.caller, auth);
+        self
+    }
+
+    pub fn deauthorize_caller(self, auth: Auth) -> Self {
+        self.as_controller().deauthorize(&self.caller, auth);
+        self
+    }
+
+    pub fn request_cost(
+        &self,
+        source: Source,
+        json_rpc_payload: &str,
+        max_response_bytes: u64,
+    ) -> Nat {
+        self.call_query(
+            "request_cost",
+            Encode!(&source, &json_rpc_payload, &max_response_bytes).unwrap(),
+        )
+    }
+}
+
+#[test]
+fn test_register_provider() {
+    let setup = EvmRpcSetup::new().authorize_caller(Auth::RegisterProvider);
+
+    let a_id = setup.register_provider(RegisterProviderArgs {
+        chain_id: 1,
+        hostname: "cloudflare-eth.com".to_string(),
+        credential_path: "".to_string(),
+        credential_headers: None,
+        cycles_per_call: 0,
+        cycles_per_message_byte: 0,
+    });
+    let b_id = setup.register_provider(RegisterProviderArgs {
+        chain_id: 5,
+        hostname: "ethereum.publicnode.com".to_string(),
+        credential_path: "".to_string(),
+        credential_headers: None,
+        cycles_per_call: 0,
+        cycles_per_message_byte: 0,
+    });
+    assert_eq!(a_id + 1, b_id);
+    let providers = setup.get_providers();
+    assert_eq!(
+        providers[providers.len() - 2..],
+        vec![
+            ProviderView {
```

Comment:
Maybe not necessary but how about implementing the `From` trait for `RegisterProviderArgs -> ProviderView`?

Reply:
As a slight variation of this idea, I added `From<Provider> for ProviderView`, which simplifies code elsewhere in the repository. 

Converting `RegisterProvider` to `ProviderView` is a bit trickier because `ProviderView.provider_id` is assigned after registering the provider.